packages/jupyter-ai/jupyter_ai/actors/chat_provider.py

```diff
@@ -0,0 +1,42 @@
+from jupyter_ai.actors.base import Logger, ACTOR_TYPE
+from jupyter_ai.models import ProviderConfig
+from jupyter_ai_magics.utils import decompose_model_id
+import ray
+
+@ray.remote
+class ChatProviderActor():
+
+    def __init__(self, log: Logger):
+        self.log = log
+        self.provider = None
+        self.provider_params = None
+
+    def update(self, config: ProviderConfig):
+        providers_actor = ray.get_actor(ACTOR_TYPE.PROVIDERS.value)
+        o = providers_actor.get_model_providers.remote()
+        providers = ray.get(o)
+        provider_id, local_model_id = decompose_model_id(model_id=config.model_provider, providers=providers)
+        
+        p = providers_actor.get_model_provider.remote(provider_id)
+        provider = ray.get(p)
+        
+        if not provider:
+            raise ValueError(f"No provider and model found with '{config.model_provider}'")
+        
+        auth_strategy = provider.auth_strategy
+        api_keys = config.api_keys
+        if auth_strategy:
+            if auth_strategy.type == "env" and auth_strategy.name.lower() not in api_keys:
+                raise ValueError(f"Missing value for '{auth_strategy.name}' in the config.")
+            
+            provider_params = { "model_id": local_model_id}
+            api_key_name = auth_strategy.name.lower()
+            provider_params[api_key_name] = api_keys[api_key_name]
+            self.provider = provider
+            self.provider_params = provider_params
```

Comment:
Don't think the control block is correct; it's not setting the attributes in the case `auth_strategy is None`. It also incorrectly assumes all `auth_strategy` instances are of type `EnvAuthStrategy` (not true for SM Endpoints). Instead, I would prefer:

```py
provider_params = { "model_id": local_model_id }

if auth_strategy and auth_strategy.type == "env":
  name = auth_strategy.name.lower()
  if name not in api_keys:
    raise ValueError(f"Missing value for '{auth_strategy.name}' in the config.")
  provider_params[name] = api_keys[name]

self.provider = provider
self.provider_params = provider_params
```

Reply:
Good catch. Thanks.